framework/src/modules/chain/logic/block.js

```diff
@@ -106,11 +109,13 @@ class Block {
 		let size = 0;
 
 		const blockTransactions = [];
-		const payloadHash = crypto.createHash('sha256');
+		const payloadHash = deepHashBuffer(
+			transactions.map(transaction => transaction.getBytes())
+		);
 
 		for (let i = 0; i < transactions.length; i++) {
 			const transaction = transactions[i];
-			const bytes = transaction.getBytes(transaction);
+			const bytes = transaction.getBytes();
```

Comment:
const bytes = transactionsBytes[i];

Reply:
probably this part needs to be cleaned up in other issue